### PR TITLE
Feature/contact modal states

### PR DESCRIPTION
**Pull request by eugenialcaraz**
#### Trello board reference:

* [5](Create Contact Modal)

---

#### Description:

* Add success and error view after submitting a message
* Add error message & styling

---

#### Reviewers:

* @nicmalegre 
* @jpirazusta 
* @tobicorradi 
* @diego317 

---

#### Notes:

* Message is still not submitted through the API, functionality will be covered in a separated ticket

---

#### Preview:

image

image

image

### CODE REVIEW

File: src/components/common/forms/ContactForm.tsx [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
 const ContactForm = () => {
-  const { register, handleSubmit } = useForm();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const dispatch = useAppDispatch();
+  const formError = errors["email*"]?.type || errors["message*"]?.type;
```
Comment (nicmalegre) [inline src/components/common/forms/ContactForm.tsx]: Are you using this only to handle if there is some error but you don’t need to know which input has the error right? You could do something like
` const isFormValid = Object.keys(errors).length === 0;`

File: src/components/common/forms/ContactForm.tsx [hunk 2]
```diff
@@ -1,28 +1,47 @@
...
+
   const onSubmit = (data: object) => {
     //on submit logic
```
Comment (nicmalegre) [inline src/components/common/forms/ContactForm.tsx]: Is this comment necessary?

File: src/components/common/forms/ContactForm.tsx (hunk 1 above)
Comment (eugenialcaraz) [reply src/components/common/forms/ContactForm.tsx]: It looks cleaner, thanks! I'll change it

File: src/components/common/forms/ContactForm.tsx (hunk 2 above)
Comment (eugenialcaraz) [reply src/components/common/forms/ContactForm.tsx]: Not really! It was mostly for your reference. I've erased it now

**Review by nicmalegre — APPROVED** [review]

File: src/components/common/forms/ContactForm.tsx [hunk 3]
```diff
@@ -1,28 +1,46 @@
...
 const ContactForm = () => {
-  const { register, handleSubmit } = useForm();
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
+
+  const dispatch = useAppDispatch();
+  const isFormValid = Object.keys(errors).length === 0;
```
Comment (jpirazusta) [inline src/components/common/forms/ContactForm.tsx]: Is this needed? I think you could use `isValid`, from `formState`

**Review by jpirazusta — APPROVED** [review]

File: src/components/common/forms/ContactForm.tsx (hunk 3 above)
Comment (eugenialcaraz) [reply src/components/common/forms/ContactForm.tsx]: I like it! I also added `isSubmitted` , as `isValid` starts in `false`, and I need to show the initial inputs without error if they hasn't been submitted

```
   <span className={isSubmitted ? (isValid ? "" : styles.error) : ""}>
        Both fields are mandatory
      </span>
```

File: src/components/common/forms/ContactForm.tsx [hunk 4]
```diff
@@ -11,11 +11,11 @@ const ContactForm = () => {
   const {
     register,
     handleSubmit,
-    formState: { errors },
+    formState: { isValid, isSubmitted },
   } = useForm();
 
   const dispatch = useAppDispatch();
-  const isFormValid = Object.keys(errors).length === 0;
+  // const isFormValid = Object.keys(errors).length === 0;
```
Comment (jpirazusta) [inline src/components/common/forms/ContactForm.tsx:18]: Remember to remove this commented line

Comment (jpirazusta) [reply src/components/common/forms/ContactForm.tsx:18]: I see you're always using the same logic for checking the validity. You could add a const for that:
`const isFormValid = isSubmitted && isValid;`

Comment (eugenialcaraz) [reply src/components/common/forms/ContactForm.tsx:18]: Lovit, I'll apply it in the opened PR

**Pull request merged**